### ISSUE TITLE
Keypad inserts space rather than executing command

### ISSUE THREAD

**Issue by ircurry** [meow-edit/meow#691]
I recently updated meow to the latest commit (73f2dd2) and found that the keypad does not work as intended. When I press a key while in keypad instead of executing the command bound the that key in `meow-keypad-state-map` the space character is inserted. The key still seems to be read (for instance dashboard `SPC p` will jump to the project section as if I had only pressed `p`). 

I found that the commit that introduced this was 8ae8b2c. My emacs configuration [can be found here](https://github.com/ircurry/cfg/blob/da566c4ba46ab2fe5d74424ff9c60a6647fd3025/home/programs/emacs/emacs.org). I have also attached a video of the behavior. The video quality is poor for some reason but I think it suffices to show the issue. The specific version of meow in the video is on master 8ae8b2c.

https://github.com/user-attachments/assets/76f8d530-09d0-407c-936d-27f83a3e6e84

If there is anything else I can do to get this resolved please let me know.

**Comment by DogLooksGood** [comment]
Hey, try `(setq meow-keypad-leader-dispatch "C-c")`

**Comment by ircurry** [comment]
That worked like a charm. Issue completely went away. Why did that work?

**Comment by erganemic** [comment]
Not sure if this is the same issue, but I'm experiencing something similar; I'm binding `C-c RET` for a common command:

```
; I also have (meow-leader-define-key '("u" . meow-universal-argument)) in my use-package :config block
(use-package gptel
  :ensure t
  :bind* (("C-c RET" . gptel-send)) ; bind in override-global-map for all modes
  :bind (("C-c RET" . gptel-send))) ; Also bind globally in mode-specific-map for meow
```

I call this most frequently as `SPC u SPC RET` and occasionally, instead of calling 'gptel-send with a prefix argument, this inserts 4 newlines, like I'd expect if I just called `SPC u RET`. Weirdly, this seems to happen randomly; I thought it might be context-sensitive keymaps, but nope, I can do `SPC u SPC RET` and get newlines, and then if I navigate back to the same spot and do it again, I'll get `gptel-send with a prefix argument, as expected.

Since it's hard to reproduce, I'll set this var and see if I experience it again, but in the mean time, if anyone has any idea why keybindings would behave differently stochastically, let me know--I feel like I'm going crazy!

**Comment by ircurry** [comment]
I seem to have a different problem now that I have done `(setq meow-keypad-leader-dispatch "C-c")`. The buffer local bindings under `C-c` overwrite what I have set in my keypad. I prefer to have everything under my leader key be universal and work the same no matter the buffer. I also found that setting `meow-keypad-leader-dispatch` to a sparse keymap creates the same behavior as when set to nil.

**Comment by ircurry** [comment]
From the docstring of `meow-leader-define-key`: 

> Meow use ‘mode-specific-map’ as leader keymap.
> Usually, the command on C-c <key> can be called in Meow via SPC <key>.
> 
> Thus, users should not add a dispatching keybinding like ("<key>" . "C-c <key>")
> with this helper, it will result in recursive calls.
It seems that this is why. `meow-leader-define-key` effectively just defines the default bindings that get overridden by the local map. With the way this is headed, it looks like I'm going to have to bind everything to a keymap normally which would be a pain since I use binding to a keychord (e.g. `'(f . "C-x C-f")` extensively. This was nice for opt-in local bindings. Is there anyone who knows of a way around this?

**Comment by ircurry** [comment]
@erganemic are you able to consistently reproduce the issue by waiting for the which-key pop up to pop up?

**Comment by erganemic** [comment]
Yep, it consistently inserts newlines instead of calling the desired command if I wait for the popup to appear. Nice to have that "why" answered :grin:

**Comment by ircurry** [comment]
Awesome! Hopefully the latest commit will fix the issue for you too. It also makes sense why it was doing it for me every time since I have which key set to be near instant.
```emacs-lisp
(use-package which-key
  :demand t
  :init (which-key-mode)
  :diminish which-key-mode
  :config
  (setq which-key-idle-delay 0.0001))
```

**Comment by DogLooksGood** [comment]
This issue should be fixed at 57e2c29fbdc5108e43eff5d4fb11423c1d325ab2. 

Now it should work with all leader dispatch types.

**Comment by ircurry** [comment]
Awesome, thank you. Closing the issue now.
